middleware/kubernetes/k8sclient/k8sclient.go

```diff
@@ -23,13 +25,17 @@ type K8sConnector struct {
 }
 
 func (c *K8sConnector) SetBaseUrl(u string) error {
-    validUrl, error := url.Parse(u)
+    url, error := url.Parse(u)
 
     if error != nil {
         return error
     }
-    c.baseUrl = validUrl.String()
 
+	if ! url.IsAbs() {
+		return errors.New("k8sclient: Kubernetes endpoint url must be an absolute URL")
+	}
+
+    c.baseUrl = url.String()
```

Comment:
http.Get() takes a string rather than a Url object. I am trying to avoid the conversion of Url -> string for each API call. However, I don't feel strongly about this so I'm happy to change to storing a Url.